src/vcpkg/vcpkgcmdarguments.cpp

```diff
@@ -388,16 +388,44 @@ namespace vcpkg
         if (const auto p = args.debug_env.get()) args.debug = *p;
 
         auto maybe_command = args.parser.extract_first_command_like_arg_lowercase();
+        args.forwardable_arguments = args.parser.get_remaining_args();
+        bool use_stderr = Util::contains(args.forwardable_arguments, "--x-json");
```

Comment:
Instead of hard coding stuff like this, can we add a field to CommandMetadata that controls the setting?

Alternately, can we just do it at the top of the command function in question instead?

Reply:
The problem then it is too late. There are Debug print statements before a command is selected.

Reply:
For example here:
https://github.com/microsoft/vcpkg-tool/blob/e658895c8b03f64f4176ced5d16063a451c028ce/src/vcpkg.cpp#L279-L292
After line 283 we already have to know it output should go to stderr or stdout

Reply:
But one could argue that we don't care if debug output breaks things

Reply:
I would like to resolve those as 'debug output always goes to stderr' and 'console parse failures always go to stderr' on the grounds that neither of those types of output should apply to the edge cases @ras0219-msft enumerated above; I'm asking around and will keep you posted.